Source/WebKit/UIProcess/API/glib/WebKitWebContext.cpp

```diff
@@ -2023,7 +2023,7 @@ WebProcessPool& webkitWebContextGetProcessPool(WebKitWebContext* context)
 
 void webkitWebContextCreatePageForWebView(WebKitWebContext* context, WebKitWebView* webView, WebKitUserContentManager* userContentManager, WebKitWebView* relatedView, WebKitWebsitePolicies* defaultWebsitePolicies)
 {
-    auto pageConfiguration = API::PageConfiguration::create();
+    auto pageConfiguration = relatedView ? webkitWebViewGetConfigurationForNextRelatedView(relatedView) : API::PageConfiguration::create();
     pageConfiguration->setProcessPool(context->priv->processPool.get());
     pageConfiguration->setPreferences(webkitSettingsGetPreferences(webkit_web_view_get_settings(webView)));
     pageConfiguration->setRelatedPage(relatedView ? &webkitWebViewGetPage(relatedView) : nullptr);
```

Comment:
Can webkitWebContextCreatePageForWebView be called with a null relatedView?

I think these page configuration setters are duplicate work if relatedView is non-null because the values area already set in WebPageProxy::createNewPage, but I don't think they hurt.  I also can't say I'm completely following all possible code flows here.

Reply:
Reviewing this I've realized that page configuration creation doesn't really belong to web context, so I've created https://github.com/WebKit/WebKit/pull/28383 to move it to the web view which will simplify this patch too.

Reply:
> Can webkitWebContextCreatePageForWebView be called with a null relatedView?

Yes, and that case was broken and would trigger the assert. Good catch.